frontend/src/metabase/visualizations/lib/apply_tooltips.js

```diff
@@ -111,8 +111,8 @@ export function getClickHoverObject(
           col: col,
         };
       });
+      dimensions = rawCols.map((column, i) => ({ column, value: rawRow[i] }));
```

Comment:
The tooltip is actually taken care already with the existing code.
The move here is to ensure that no exception is thrown (since in the case of waterfall charts, `rawRow` can be undefined) and the rest of the logic continue producing the right text for the tooltip.